### ISSUE TITLE
TypeScript type error with React 19 types

### ISSUE THREAD

**Issue by danielrentz** [i18next/react-i18next#1823]
## 🐛 Bug Report

Type error with React v19 types. The global type namespace `JSX` has been removed for better compatibility with other JSX libs.

https://react.dev/blog/2024/04/25/react-19-upgrade-guide#the-jsx-namespace-in-typescript

## To Reproduce

just use `@types/react` v19 and `react-i18next` in a project, and run `tsc` (without `skipLibChecks`).

```
node_modules/react-i18next/index.d.ts:147:23 - error TS2694: Namespace 'global.JSX' has no exported member 'LibraryManagedAttributes'.

147   ResolvedProps = JSX.LibraryManagedAttributes<
                          ~~~~~~~~~~~~~~~~~~~~~~~~
```

## Expected behavior

no type error

## Your Environment

- *runtime version*: i.e. node v20.18.0
- *i18next version*: i.e. 24.1.0
- *os*: Windows 11
- *React version*: 19.0.0
- *@types/react version*: 19.0.1

**Comment by adrai** [comment]
Can you check if this addressed that issue? https://github.com/i18next/react-i18next/pull/1822

**Comment by danielrentz** [comment]
@adrai Exactly!

**Comment by adrai** [comment]
addressed in v15.2.0

**Comment by danielrentz** [comment]
@adrai works for me, thank you!
